cob_twist_controller/src/damping_methods/damping.cpp

```diff
@@ -76,10 +81,14 @@ inline double DampingNone::getDampingFactor(const Eigen::VectorXd& sorted_singul
 /**
  * Method just returns the damping factor from ros parameter server.
  */
-inline double DampingConstant::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
+inline Eigen::MatrixXd DampingConstant::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
                                                 const Eigen::MatrixXd& jacobian_data) const
 {
-    return this->params_.damping_factor;
+	uint32_t rows = sorted_singular_values.rows();
+	//uint32_t cols = sorted_singular_values.rows(); //NOt needed S matrix is always square
+	Eigen::MatrixXd damping_factor = Eigen::MatrixXd::Zero(rows,rows);
+	damping_factor(sorted_singular_values.rows() - 1,sorted_singular_values.rows() - 1) = pow(this->params_.damping_factor,2);
```

Comment:
Shouldn't this be a diagonal matrix? Currently it's a zero matrix with an entry only in ```sorted_singular_values.rows() - 1,sorted_singular_values.rows() - 1```